### PR TITLE
Enable token based auth for panel api oerations

### PR DESCRIPTION
**Pull request by pawzar**
Panel uses cookies only for `/auth`.
All other calls are made statelessly to `/panel`.

### CODE REVIEW

File: src/app/auth/auth.service.ts [hunk 1]
```diff
@@ -1,61 +1,69 @@
-import { HttpClient } from '@angular/common/http';
-import { Injectable } from '@angular/core';
-import { Observable } from 'rxjs/Observable';
+import {HttpClient} from '@angular/common/http';
+import {Injectable} from '@angular/core';
+import {Observable} from 'rxjs/Observable';
```
Comment (c3zi) [inline src/app/auth/auth.service.ts]: Do we have any ESLint configuration for the project? My configuration does not allow to use curly brackets without single space ;/

File: src/app/auth/auth.service.ts [hunk 2]
```diff
@@ -1,61 +1,69 @@
...
+  emailActivationResend(uri){
+    return this.http.post(`${environment.authUrl}/users/email/activate/resend`, {uri});
   }
+
+  // token access
+
+  resetPassword(user: object, uri: string) {
+    return this.http.patch(`${environment.apiUrl}/users`, {user, uri});
+   }
+
```
Comment (c3zi) [inline src/app/auth/auth.service.ts]: Don't we want to have a dedicated endpoint for resetting a password (e.g. `/users/reset-password`)? 
I'm thinking what happens on the backend - we have one endpoint for patching data and I guess that we use it only for resetting a password now? 
Nothing important but I wanted to point out that resetting is so specified action that maybe custom/dedicated endpoint makes sense :)

File: src/app/common/request.interceptor.ts [hunk 3]
```diff
@@ -51,11 +45,17 @@ export class RequestInterceptor implements HttpInterceptor {
...
+      request = request.clone({
+        withCredentials: true
+      });
+    } else {
+      request = request.clone({
+        setHeaders: {
+          Authorization: `Bearer ${this.session.getUser().apiToken}`
+        }
+      });
+    }
```
Comment (c3zi) [inline src/app/common/request.interceptor.ts]: We can set only parameters in the condition and call `request.clone` once. Up to you.
Btw. I tried to use one-line condition ( (condition) ? value1 : value2 ) but it is not readable  ;-)

```
 let requestParameters;
 if (request.url.startsWith(environment.authUrl)) {
      requestParameters = {
        withCredentials: true,
      };
    } else {
      requestParameters = {
        setHeaders: {
          Authorization: `Bearer ${this.session.getUser().apiToken}`,
        }
      };
    }

request = request.clone(requestParameters);  
```

File: test/pl/adshares/adpanel/setup/DriverProvider.java [hunk 4]
```diff
@@ -40,10 +41,19 @@ public static File getWebDriverFile(String name) {
 
   static WebDriver getWebDriver() {
     if (driver == null) {
-      System.setProperty("webdriver.chrome.driver", getWebDriverFile("chromedriver").getAbsolutePath());
+//      System.setProperty("webdriver.chrome.driver", getWebDriverFile("chromedriver").getAbsolutePath());
+
```
Comment (c3zi) [inline test/pl/adshares/adpanel/setup/DriverProvider.java]: We can remove this comment, it's the same as the line above ;-)

File: test/pl/adshares/adpanel/setup/DriverProvider.java [hunk 5]
```diff
@@ -40,10 +41,19 @@ public static File getWebDriverFile(String name) {
...
       ChromeOptions chromeOptions = new ChromeOptions();
-      chromeOptions.addArguments("--start-maximized");
-      driver = new ChromeDriver(chromeOptions);
+      chromeOptions.addArguments("start-maximized");
+
+try{
+driver = new RemoteWebDriver(new java.net.URL("http://localhost:4444/wd/hub"), chromeOptions);
+}catch (java.net.MalformedURLException $e){
+throw new RuntimeException($e.getMessage());
+}
```
Comment (c3zi) [inline test/pl/adshares/adpanel/setup/DriverProvider.java]: 1. Indentation
2. Could we move URL address to an environment variable?

File: src/app/common/request.interceptor.ts (hunk 3 above)
Comment (pawzar) [reply src/app/common/request.interceptor.ts]: :ballot_box_with_check:

File: src/app/auth/auth.service.ts (hunk 1 above)
Comment (pawzar) [reply src/app/auth/auth.service.ts]: :ballot_box_with_check:

File: src/app/auth/auth.service.ts (hunk 2 above)
Comment (pawzar) [reply src/app/auth/auth.service.ts]: not this branch

**Pull request closed without merging**